osu.Game/Overlays/DirectOverlay.cs

```diff
@@ -232,21 +251,49 @@ protected override void PopIn()
         private SearchBeatmapSetsRequest getSetsRequest;
 
         private readonly Bindable<string> currentQuery = new Bindable<string>(string.Empty);
+        private int currentPage = 1;
+        private bool isLastPageLoaded = false;
```

Comment:
Admittedly I'm uneasy about all this state landing in here. Feels like the overlay might get a little bloaty. This could maybe be factored out to a paged request handling component? Purely spitballing:

```csharp
class BeatmapSetPager
{
   public event PageFetched;

   // returns true if page present, false if not
   public bool FetchNextPage() { ... }
   public void ResetToFirstPage() { ... }
}
```

Does that sound feasible?